Assets/Mirror/Editor/Weaver/CompilationFinishedHook.cs

```diff
@@ -46,32 +49,29 @@ static void HandleError(string msg)
         [InitializeOnLoadMethod]
         static void OnInitializeOnLoad()
         {
-            CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
-
             // We only need to run this once per session
             // after that, all assemblies will be weaved by the event
             if (!SessionState.GetBool("MIRROR_WEAVED", false))
             {
+                // reset session flag
                 SessionState.SetBool("MIRROR_WEAVED", true);
+
                 WeaveExistingAssemblies();
             }
+
+            // hook the event after the first run of WeaveExistingAssemblies to avoid a double firing
+            CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
         }
 
-        static void WeaveExistingAssemblies()
+        public static void WeaveExistingAssemblies()
         {
             foreach (UnityAssembly assembly in CompilationPipeline.GetAssemblies())
-            {
                 if (File.Exists(assembly.outputPath))
-                {
                     OnCompilationFinished(assembly.outputPath, new CompilerMessage[0]);
-                }
-            }
-#if UNITY_2019_3_OR_NEWER
-            EditorUtility.RequestScriptReload();
-#else
-            UnityEditorInternal.InternalEditorUtility.RequestScriptReload();
-#endif
 
+            // Previously this method called UnityEditorInternal.InternalEditorUtility.RequestScriptReload
+            // but doing so now creates an endless loop of reloading and reweaving because now we're blocking
+            // play mode for weaver failures. Let Unity be Unity and don't RequestScriptReload anymore.
```

Comment:
are you sure about that part?
there must have been a good reason why RequestScriptReload was necessary

Reply:
It may have been but I know for sure if I leave that in it keeps cycling endlessly.